kornia/feature/harris.py

```diff
@@ -74,42 +85,70 @@ def forward(self, input: torch.Tensor) -> torch.Tensor:  # type: ignore
                              .format(input.shape))
         # compute the first order gradients with sobel operator
         # TODO: implement support for kernel different than three
-        gradients: torch.Tensor = spatial_gradient(input)
+        grad_mode = 'Sobel' if self.do_blur else 'NoBlur'
+        gradients: torch.Tensor = spatial_gradient(input, grad_mode)
         dx: torch.Tensor = gradients[:, :, 0]
         dy: torch.Tensor = gradients[:, :, 1]
 
         # compute the structure tensor M elements
         def g(x):
             return gaussian_blur2d(x, (3, 3), (1., 1.))
-
         dx2: torch.Tensor = g(dx * dx)
         dy2: torch.Tensor = g(dy * dy)
         dxy: torch.Tensor = g(dx * dy)
-
         det_m: torch.Tensor = dx2 * dy2 - dxy * dxy
         trace_m: torch.Tensor = dx2 + dy2
 
         # compute the response map
-        scores: torch.Tensor = det_m - self.k * trace_m ** 2
+        scores: torch.Tensor = det_m - self.k * (trace_m ** 2)
 
         # threshold
         # TODO: add as signature parameter ?
-        scores = torch.clamp(scores, min=1e-6)
+        scores = torch.clamp(scores, min=0)
 
         # apply non maxima suppresion
-        scores = non_maxima_suppression2d(scores, kernel_size=(3, 3))
+        if self.nms:
+            scores = non_maxima_suppression2d(scores, kernel_size=(3, 3))
 
         # normalize and return
-        scores_max: torch.Tensor = F.adaptive_max_pool2d(scores, output_size=1)
-        return scores / scores_max
+        if self.normalize:
+            scores_max: torch.Tensor = F.adaptive_max_pool2d(scores,
+                                                             output_size=1)
+            scores = scores / scores_max.clamp(1e-6)
+        return scores
+
+    def update_response(self, resp, sigmas):
```

Comment:
I would put it as a separated function since it doesn't use any state of sobel. Actually sobel doesn't hold state. In that case we should consider writing it as a free function.

Reply:
But it depends on being "harris-related". For other detector it would not be pow(4), etc. Having free function is fine, but module should have it as well, IMO

Reply:
We can have a mode parameter with different implementations. The module should call the free function similar to other functions we have

Reply:
As for the function name in case is standalone wouldn't be more appropriate something like refine or scale scores_map ? Is there any other library implementing it?

Reply:
rescale_scores_map is  good. Other libraries put sigma directly into to main function, which I find somewhat artificial.

Reply:
> We can have a mode parameter with different implementations. The module should call the free function similar to other functions we have

Agree. I`ll refactor this. It goes to be cleaner with your suggestion